src/metabase/integrations/ldap.clj

```diff
@@ -180,6 +189,14 @@
     (catch Exception e
       {:status :ERROR, :message (.getMessage e)})))
 
+(defn test-current-ldap-details
+  "Tests the connection to an LDAP server using the currently set settings."
+  []
+  (let [settings (into {} (map
+                           (fn [[k v]] [v (setting/get k)])
+                           mb-settings->ldap-details))]
```

Comment:
On first glance I missed that the new keys are the old values. I think this is easier to see in the following code:
```suggestion
  (let [settings (into {} (for [[k v] mb-settings->ldap-details]
                             [v (setting/get k)]))]
```